### PR TITLE
feat(apig/quota): add new resource to associate apps

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
add a new resource to manage association relationship for application quota and the applications.

CheckDeleted：

（Refresh）
image

API Debug results:
1. the instance does not exist:
image

2. one of the application, quota and the associate relationship does not exist:
image

（Delete）
image

API Debug results:
1. the instance does not exist:
image

2. one of the application, quota and the associate relationship does not exist:
image

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new resource to associate applications to quota.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/apig' TESTARGS='-run=TestAccApplicationQuotaAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apig -v -run=TestAccApplicationQuotaAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccApplicationQuotaAssociate_basic
=== PAUSE TestAccApplicationQuotaAssociate_basic
=== CONT  TestAccApplicationQuotaAssociate_basic
--- PASS: TestAccApplicationQuotaAssociate_basic (609.25s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apig      609.308s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_application_quota_associate_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/apig"
+)
+
+func getApplicationQuotaAssociateFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.NewServiceClient("apig", acceptance.HW_REGION_NAME)
+	if err != nil {
+		return nil, fmt.Errorf("error creating APIG Client: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_application_quota_associate_test.go]:
```suggestion
		return nil, fmt.Errorf("error creating APIG client: %s", err)
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+		Target:  []string{"COMPLETED"},
+		Refresh: appsUnbindingRefreshFunc(client, instanceId, quotaId, appIds),
+		Timeout: timeout,
+		// In most cases, the bind operation will be completed immediately, but in a few cases, it needs to wait
+		// for a short period of time, and the polling is performed by incrementing the time here.
+		MinTimeout: 2 * time.Second,
+	}
+	_, err := stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return fmt.Errorf("error waiting for the binding completed: %s", err)
```
Comment (deer-hang) [inline huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go]:
```suggestion
		return fmt.Errorf("error waiting for the unbinding completed: %s", err)
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+	)
+
+	client, err := cfg.NewServiceClient("apig", region)
+	if err != nil {
+		return diag.Errorf("error creating APIG client: %s", err)
+	}
+
+	oldVal, _ := d.GetChange("applications")
+	if err = disassociateAppsFromQuota(ctx, client, instanceId, quotaId, parseAssociatedAppIds(oldVal.(*schema.Set)),
+		d.Timeout(schema.TimeoutUpdate)); err != nil {
```
Comment (deer-hang) [inline huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go]:
```suggestion
		d.Timeout(schema.TimeoutDelete)); err != nil {
```

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go [hunk 4]
```diff
@@ -0,0 +1,452 @@
...
+		quotaId    = d.Get("quota_id").(string)
+	)
+
+	client, err := cfg.NewServiceClient("apig", region)
+	if err != nil {
+		return diag.Errorf("error creating APIG client: %s", err)
+	}
+
+	oldVal, _ := d.GetChange("applications")
+	if err = disassociateAppsFromQuota(ctx, client, instanceId, quotaId, parseAssociatedAppIds(oldVal.(*schema.Set)),
```
Comment (deer-hang) [inline huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go:428]: Please precheck `oldVal.(*schema.Set).Len() > 0`

File: huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_application_quota_associate_test.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/services/acceptance/apig/resource_huaweicloud_apig_application_quota_associate_test.go]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go (hunk 3 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go]: done.

File: huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go (hunk 4 above)
Comment (Lance52259) [reply huaweicloud/services/apig/resource_huaweicloud_apig_application_quota_associate.go:428]: There is unnecessary to check the length of old set. In the delete phase, if the length of old set is zero, checkdeleted logic will be triggered in the refresh step before enter delete step.

**Pull request merged**